### ISSUE TITLE
Policy assignment API version upgrade

### ISSUE THREAD

**Issue by tohov** [Azure/terraform-azurerm-caf-enterprise-scale#285]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Versions

**terraform**: 1.1.6

**azure provider**: 2.98.0

**module**: 1.1.3

### Description

**Describe the bug**

Could you update the policy assignment API version from 2019-09-01 to the newer or latest version or is there a possibility to introduce unwanted breaking change? I can see that the policy definitions are already using newer API version, so maybe the API version upgrade is already on roadmap for assignments as well?

There's some new properties that would be useful, e.g. the non-compliance messages.

**Comment by krowlandson** [comment]
Hi @tohov... we currently support backwards compatibility to `v2.77.0` of the provider which pre-dates the recent addition of support for non-compliance messages (added by our very own @matt-FFFFFF)

We have other reasons to update this in the next release though so I will see if we can include this as a part of that (or the one after).

**Comment by krowlandson** [comment]
Release `v2.0.0` is now live, with detailed instructions regarding the upgrade also available.

Please see the release notes for more information:
https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/releases/tag/v2.0.0

With this update we can now start planning to resolve this issue so please look out for updates in the coming weeks.

**Comment by matt-FFFFFF** [comment]
This is now unblocked but is a duplicate of #67 

Will close this and track there.
